src/Facebook/Url/FacebookUrlDetectionHandler.php

```diff
@@ -160,4 +160,23 @@ protected function getHeader($key)
     {
         return $this->getServerVar('HTTP_' . $key);
     }
+
+    /**
+     * Checks if the value in X_FORWARDED_HOST is a valid hostname
+     * Could prevent unintended redirections
+     *
+     * @param string $header
+     *
+     * @return boolean
+     */
+    protected function isValidForwardedHost($header)
+    {
+        $elements = explode(',', $header);
+        $host = $elements[count($elements) - 1];
+        
+        return preg_match("/^([a-z\d](-*[a-z\d])*)(\.([a-z\d](-*[a-z\d])*))*$/i", $host) //valid chars check
+            && preg_match("/^.{1,253}$/", $host) //overall length check
```

Comment:
looks like you forgot this comment: `&& 0 < strlen($host) && strlen($host) < 254` looks clearer to me, and avoid the cost of a `preg_match` for length tests